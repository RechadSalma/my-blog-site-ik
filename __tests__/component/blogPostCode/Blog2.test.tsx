import { render } from "@testing-library/react";
import Blog1Codes from "../../../components/blogPostCodes/Blog2";

describe("<Blog2/> code for blog2 component", () => {
  const blogOneComponent = () => render(<Blog1Codes />);

  test("renders the <code> tag", () => {
    const { getByRole } = blogOneComponent();

    const codeTag = getByRole("code");

    expect(codeTag).toBeInTheDocument();
  });

  // snapshot component
  test("snapshot Blog2 code tag", () => {
    const { container } = blogOneComponent();

    expect(container).toMatchSnapshot();
  });
});
