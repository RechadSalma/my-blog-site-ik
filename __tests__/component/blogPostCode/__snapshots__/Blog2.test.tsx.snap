// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Blog2/> code for blog2 component snapshot Blog2 code tag 1`] = `
<div>
  <pre
    class="preCode mx-auto"
  >
    <code
      class="codeTag mx-auto"
      role="code"
    >
      <div>
        <span
          class="comment"
        >
          // eg. sqlTimestamp: 2021-06-01T11:08:01.000Z
        </span>
      </div>
      <div>
        <span
          class="keyword"
        >
          function
        </span>
        (sqlTimestamp) {
      </div>
      <div
        class="tab1"
      >
        <span
          class="keyword"
        >
          const
        </span>
         removeT = sqlTimestamp
        <span
          class="method"
        >
          .replace
        </span>
        (
        <span
          class="value"
        >
          "T"
        </span>
        , 
        <span
          class="value"
        >
          " "
        </span>
        );
      </div>
      <br />
      <div
        class="tab1"
      >
        <span
          class="keyword"
        >
          const
        </span>
         removeMilliSecondRegEx = /\\.\\w+/g;
      </div>
      <br />
      <div
        class="tab1"
      >
        <span
          class="keyword"
        >
          const
        </span>
         timestamp = removeT
        <span
          class="method"
        >
          .replace
        </span>
        (removeMilliSecondRegEx, 
        <span
          class="value"
        >
          ""
        </span>
        ); 
      </div>
      <br />
      <div
        class="tab1"
      >
        <span
          class="keyword"
        >
          return
        </span>
         timestamp
      </div>
      <div>
        }
      </div>
      <div
        class="comment"
      >
        // Output: "2021-06-01 11:08:01"
      </div>
    </code>
  </pre>
</div>
`;
